src/Model/Behavior/SocialBehavior.php

```diff
@@ -276,19 +277,17 @@ public function generateUniqueUsername($username)
      * Prepare a query to retrieve existing entity for social login
      *
      * @param \Cake\ORM\Query\SelectQuery $query The base query.
-     * @param array $options Find options with email key.
+     * @param mixed $email Find options with email key.
      * @return \Cake\ORM\Query\SelectQuery
      */
-    public function findExistingForSocialLogin(\Cake\ORM\Query\SelectQuery $query, array $options)
+    public function findExistingForSocialLogin(SelectQuery $query, mixed $email): SelectQuery
```

Comment:
```suggestion
    public function findExistingForSocialLogin(SelectQuery $query, ?string $email = null): SelectQuery
```

Reply:
even though it's questionable if it should be nullable in the first place.

Reply:
I applied your change, thanks